webpages/settings/index.js

```diff
@@ -154,6 +154,18 @@ const AddonBody = Vue.extend({
 
       return matchesTag && matchesSearch && matchesEasterEgg;
     },
+    addonMatchesCategory() {
+      console.log(this.addon._addonId, "just ran!");
+      const matchesEnabled =
+        this.category.enabledAddon === undefined || this.category.enabledAddon === this.addon._enabled;
+      if (this.category.tagMatch) {
+        return matchesEnabled && !this.category.tagMatch.some((tag) => !this.addon.tags.includes(tag));
+      }
+      if (this.category.noTagMatch) {
+        return matchesEnabled && !this.category.noTagMatch.some((tag) => this.addon.tags.includes(tag));
+      }
+      return matchesEnabled;
+    },
```

Comment:
We're matching every addon against every category, which is very inefficient. We should know in advance which addons need to be in which category, instead of the other way

Reply:
It's also unnecessary to check whether an addon matches the category after load. If an addon was initially enabled then the page loaded, it should stay under "enabled addons" to avoid poofing.